[@types/react-native] Cant assign ColorValue from object

`ColorValue` was introduced in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45963
```
declare const OpaqueColorValue: unique symbol;
type OpaqueColorValue = typeof OpaqueColorValue;

export type ColorValue = string | OpaqueColorValue;
```

However, this now breaks if they are assigned to an object:
```
function test(): ColorValue {
  return "foo";
}

const colors = {
  foo: test(),
};

const styles = StyleSheet.create({
  checked: {
    color: colors.foo,
  },
});
```

Results in:
```
error TS2322: Type 'string | symbol' is not assignable to type 'string | unique symbol | undefined'.
  Type 'symbol' is not assignable to type 'string | unique symbol | undefined'.
```

Authors:
@alexbrazier 

Reviewers:
@zoontek
@rickhanlonii
@dotconnor
@TheSavior
@pesoklp13
@alloy
@tom-un

Version:
typescript: 3.9.7
@types/react-native: 0.63.8

--- a/types/react-native/index.d.ts
+++ b/types/react-native/index.d.ts
@@ -6037,9 +6037,7 @@ interface PlatformWebStatic extends PlatformStatic {
     OS: 'web';
 }
 
-declare const OpaqueColorValue: unique symbol;
-type OpaqueColorValue = typeof OpaqueColorValue;
-
+type OpaqueColorValue = symbol & { __TYPE__: 'Color' };
 export type ColorValue = string | OpaqueColorValue;
 
 export type ProcessedColorValue = number | OpaqueColorValue;

--- a/types/react-native/test/index.tsx
+++ b/types/react-native/test/index.tsx
@@ -1521,7 +1521,7 @@ StyleSheet.create({
 });
 
 function someColorString(): ColorValue {
-    return "#000000";
+    return '#000000';
 }
 
 function somePlatformColor(): ColorValue {
@@ -1531,7 +1531,7 @@ function somePlatformColor(): ColorValue {
 const colors = {
     color: someColorString(),
     backgroundColor: somePlatformColor(),
-}
+};
 
 StyleSheet.create({
     labelCell: {
@@ -1756,10 +1756,8 @@ const I18nManagerTest = () => {
 };
 
 // LayoutAnimations
-LayoutAnimation.configureNext(LayoutAnimation.create(
-    123,
-    LayoutAnimation.Types.easeIn,
-    LayoutAnimation.Properties.opacity,
-));
+LayoutAnimation.configureNext(
+    LayoutAnimation.create(123, LayoutAnimation.Types.easeIn, LayoutAnimation.Properties.opacity),
+);
 
 LayoutAnimation.configureNext(LayoutAnimation.create(123, 'easeIn', 'opacity'));

--- a/types/react-native/test/index.tsx
+++ b/types/react-native/test/index.tsx
@@ -1520,6 +1520,34 @@ StyleSheet.create({
     },
 });
 
+function someColorString(): ColorValue {
+    return "#000000";
+}
+
+function somePlatformColor(): ColorValue {
+    return PlatformColor('test');
+}
+
+const colors = {
+    color: someColorString(),
+    backgroundColor: somePlatformColor(),
+}
+
+StyleSheet.create({
+    labelCell: {
+        color: colors.color,
+        backgroundColor: colors.backgroundColor,
+    },
+});
+
+const OpaqueTest3 = () => (
+    <View
+        style={{
+            backgroundColor: colors.backgroundColor,
+        }}
+    />
+);
+
 // ProgressBarAndroid
 const ProgressBarAndroidTest = () => {
     <ProgressBarAndroid animating color="white" styleAttr="Horizontal" progress={0.42} />;

